src/storage/data_table.cpp

```diff
@@ -1088,6 +1088,26 @@ column_map_type DataTable::GetDefaultLayout() const{
   return default_partition_;
 }
 
+void DataTable::AddTrigger(commands::Trigger new_trigger) {
+  trigger_list.AddTrigger(new_trigger);
+}
+
+int DataTable::GetTriggerNumber() {
+  return trigger_list.GetTriggerListSize();
+}
+
+commands::Trigger* DataTable::GetTriggerByIndex(int n) {
+  if (trigger_list.GetTriggerListSize() <= n)
```

Comment:
The corner case handling again needs to consider n<0, and maybe some LOG print or exception throwing would be helpful for debugging.